snowflake-api/src/responses.rs

```diff
@@ -118,33 +134,33 @@ pub struct RenewSessionResponseData {
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct QueryExecResponseData {
-    pub parameters: Vec<NameValueParameter>,
-    pub rowtype: Vec<ExecResponseRowType>,
+    // pub parameters: Vec<NameValueParameter>,
+    pub rowtype: Option<Vec<ExecResponseRowType>>,
     // default for non-SELECT queries
     // GET / PUT has their own response format
     pub rowset: Option<serde_json::Value>,
     // only exists when binary response is given, eg Arrow
     // default for all SELECT queries
     // is base64-encoded Arrow IPC payload
     pub rowset_base64: Option<String>,
-    pub total: i64,
-    pub returned: i64,    // unused in .NET
-    pub query_id: String, // unused in .NET
-    pub database_provider: Option<String>,
-    pub final_database_name: Option<String>, // unused in .NET
-    pub final_schema_name: Option<String>,
-    pub final_warehouse_name: Option<String>, // unused in .NET
-    pub final_role_name: String,              // unused in .NET
+    // pub total: i64,
```

Comment:
Why are all these commented out? Is it intended? If so, could you please add a comment explaining it and then remove the commented out code?

Reply:
> Why are all these commented out? Is it intended? If so, could you please add a comment explaining it and then remove the commented out code?

I commented out since these are not used anywhere, in their code or by us. I can bring back and make them Option so they don't error on deserialize async response